internal/armadactl/scheduling_control.go

```diff
@@ -0,0 +1,119 @@
+package armadactl
+
+import (
+	"fmt"
+
+	"github.com/armadaproject/armada/internal/common/slices"
+	"github.com/armadaproject/armada/pkg/api"
+	"github.com/armadaproject/armada/pkg/client/queue"
+)
+
+func (a *App) cordonQueue(q *api.Queue) error {
+	q.SchedulingPaused = true
+	newQueue, err := queue.NewQueue(q)
+	if err := a.Params.QueueAPI.Update(newQueue); err != nil {
+		return fmt.Errorf("error updating queue %s: %s", q.Name, err)
+	}
+	return err
+}
+
+func (a *App) CordonQueue(queue string, dryRun bool) error {
+	selectedQueue, err := a.getQueueAsAPIQueue(queue)
+	if err != nil {
+		return fmt.Errorf("error retrieving queue %s: %s", queue, err)
+	}
+
+	if dryRun {
+		fmt.Printf("Pausing scheduling for the following queues: (DRY RUN)")
+		fmt.Println(queue)
+	} else {
+		fmt.Println("Pausing scheduling for the following queues:")
+		err = a.cordonQueue(selectedQueue)
+		if err != nil {
+			return fmt.Errorf("could not pause scheduling on queue %s: %s", queue, err)
+		} else {
+			fmt.Printf("%s paused\n", queue)
+		}
+	}
+	return nil
+}
+
+func (a *App) CordonQueues(matchQueues []string, matchLabels []string, dryRun bool, inverse bool) error {
```

Comment:
how slow is this to cordon hundreds of queues?  Is it worth having an endpoint where you can cordon a large number of queues in a single call?